controllers/serviceaccount/serviceaccount_controller.go

```diff
@@ -0,0 +1,188 @@
+	r.log.V(1).Info("start reconcile", "Name", req.NamespacedName)
+	serviceAccount := &corev1.ServiceAccount{}
+
+	if r.HCPNamespace == "" {
+		return ctrl.Result{}, nil
+	}
+
+	serviceAccountExists := false
+	// Getting the hosted cluster service account
+	apiContext, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+	err := r.Get(
+		apiContext,
+		types.NamespacedName{Name: constants.MintServiceAccountName, Namespace: constants.HLFWatchedNamespace},
+		serviceAccount,
+	)
+
+	if err != nil && errors.IsNotFound(err) {
+		serviceAccountExists = false
+	} else if err == nil {
+		serviceAccountExists = true
+	} else {
+		return ctrl.Result{RequeueAfter: constants.TokenRefreshDuration}, nil
+	}
+
+	if serviceAccountExists {
+		r.log.V(1).Info("Found existing service account", "Name", serviceAccount.GetName())
+	} else {
+		serviceAccount := &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.MintServiceAccountName,
+				Namespace: constants.HLFWatchedNamespace,
+			},
+		}
+
+		// Create the service account
+		apiContext, cancel := context.WithTimeout(ctx, 10*time.Second)
+		defer cancel()
+
+		if err := r.Create(apiContext, serviceAccount); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	// create service account token
+	token, err := r.createMintServiceAccountToken(ctx, serviceAccount)
+	if err != nil {
+		r.log.Error(err, "failed to create service account token")
+		return ctrl.Result{}, err
+	}
+	r.log.V(1).Info("Refresh token", "token", token)
+
+	//copy token to secret
+	if err := r.updateOrCreateCloudWatchSecret(ctx, token); err != nil {
+		r.log.Error(err, "failed to create secret")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{RequeueAfter: constants.TokenRefreshDuration}, nil
+}
+
+func (r *ServiceAccountReconciler) createMintServiceAccountToken(
+	ctx context.Context,
+	serviceAccount *corev1.ServiceAccount,
+) (refreshToken string, err error) {
+
+	treq := &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			Audiences: []string{"openshift"},
+		},
+	}
+
+	// Create the service account token
+	apiContext, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	token, err := r.ClientSet.CoreV1().ServiceAccounts(serviceAccount.GetNamespace()).CreateToken(apiContext, serviceAccount.GetName(), treq, metav1.CreateOptions{})
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
+			r.log.V(1).Info("Token already exists", "Name", token.GetName())
+		} else {
+			r.log.Error(err, "failed to create token")
+			return "", fmt.Errorf("failed to create token: %w", err)
+		}
+	}
+
+	return token.Status.Token, nil
+}
+
+func (r *ServiceAccountReconciler) updateOrCreateCloudWatchSecret(
+	ctx context.Context,
+	token string,
+) error {
+
+	r.log.V(1).Info("Creating secrets with token")
+	ocmCloudwatchSecret := &corev1.Secret{}
+	cloCloudwatchSecret := &corev1.Secret{}
```

Comment:
```suggestion
	var ocmCloudwatchSecret, cloCloudWatchSecret &corev1.Secret{}
```

Reply:
> 

I think it may need to define as follows 
```
	var ocmCloudwatchSecret, cloCloudwatchSecret = &corev1.Secret{}, &corev1.Secret{}
```